Assets/_Functionalities/Wms/Scripts/WMSImportAdapter.cs

```diff
@@ -0,0 +1,352 @@
+            string format = GetParamValueFromSourceUrl(sourceUrl, "format");
+            format = Uri.UnescapeDataString(format);
+            if (format != "image/png" && format != "image/jpeg")
+                format = "image/png";
+            uriBuilder.AddQueryParameter("format", format);
+            if (!sourceUrl.Contains("transparent="))
+                uriBuilder.AddQueryParameter("transparent", layerPrefab.TransparencyEnabled.ToString());
+            return uriBuilder;
+        }
+
+        private class WMS
+        {
+            private readonly string sourceUrl;
+            private readonly string cachedBodyContent;
+
+            private XmlDocument xmlDocument;
+            private XmlNamespaceManager namespaceManager;
+
+            public RequestType requestType;
+            public enum RequestType
+            {
+                GetCapabilities,
+                GetMap,
+                Unsupported
+            }
+
+            public WMS(string sourceUrl, string cachedBodyFilePath)
+            {
+                this.sourceUrl = sourceUrl;
+                this.cachedBodyContent = cachedBodyFilePath;
+            }
+
+            public void ParseBodyAsXML()
+            {
+                this.xmlDocument = new XmlDocument();
+                this.xmlDocument.Load(this.cachedBodyContent);
+                this.namespaceManager = ReadNameSpaceManager(this.xmlDocument);
+            }
+
+            public bool IsGetMapRequest()
+            {
+                var getMapRequest = this.sourceUrl.ToLower().Contains("request=getmap");
+                requestType = RequestType.GetMap;
+                return getMapRequest;
+            }
+
+            public bool IsGetCapabilitiesRequest()
+            {
+                var getCapabilitiesRequest = this.sourceUrl.ToLower().Contains("request=getcapabilities");
+                requestType = RequestType.GetCapabilities;
+                return getCapabilitiesRequest;
+            }
+
+            public bool HasBboxFilterCapability()
+            {
+                return WMSBboxFilterCapability(this.xmlDocument, this.namespaceManager);
+            }
+
+            public string GetWMSVersion()
+            {
+                if (xmlDocument == null)
+                    ParseBodyAsXML();
+
+                // Use XPath to select the root node and get the version attribute
+                var rootNode = xmlDocument.SelectSingleNode("/*");
+
+                // Check if the root node is found and retrieve the version attribute
+                if (rootNode != null && rootNode.Attributes != null)
+                {
+                    var versionAttribute = rootNode.Attributes["version"];
+                    return versionAttribute?.Value; // Return the version value or null if not found
+                }
+
+                return null; // Return null if root node or version attribute is not found
+            }
+
+            public struct WMSLayerQueryParams
```

Comment:
Kan deze struct voor het overzicht in een aparte file?

Reply:
Ik zou structs nooit in eigen files zetten tenzij ze echt door meerdere controllers gebruikt worden. In dit geval ben ik vergeten de struct te verhuizen naar de adapter en hoort het een private struct te zijn omdat hij alleen bij wmsimportadapter hoort. goed gezien trouwens